### PR TITLE
Changes to Set Woocommerce Payment Method to 'Paypal' only if transac…

### PR DESCRIPTION
**Pull request by bhushan-mehta**
…tionid is exist and Payment method is Paypal

### CODE REVIEW

File: connect.php [hunk 1]
```diff
@@ -1593,8 +1593,9 @@ public function sync() {
 
 						if ( $ordercontent->paymentstatus == 'complete' ) {
 							$transaction_id = (string)$ordercontent->orderpayments[0]->orderpayment->transactionid;
+							$paymentmethod = (string)$ordercontent->orderpayments[0]->orderpayment->paymentmethod;
 
-							if ( $transaction_id ) {
+							if ( $transaction_id  && $paymentmethod == "paypal") {
```
Comment (jschiller) [inline connect.php:1598]: Not quite. We need a regular expression here as we may send "paypal" or "paypalec", so please make this match on regular expression containing paypal.

File: connect.php [hunk 2]
```diff
@@ -1652,8 +1653,9 @@ public function sync() {
...
```
Comment (jschiller) [inline connect.php:1658]: Same as above

File: connect.php (hunk 1 above)
Comment (bhushan-mehta) [reply connect.php]: Regex expression added

File: connect.php (hunk 2 above)
Comment (bhushan-mehta) [reply connect.php]: Regex expression added

File: connect.php [hunk 3]
```diff
@@ -1593,8 +1593,9 @@ public function sync() {
 
 						if ( $ordercontent->paymentstatus == 'complete' ) {
 							$transaction_id = (string)$ordercontent->orderpayments[0]->orderpayment->transactionid;
+							$paymentmethod = (string)$ordercontent->orderpayments[0]->orderpayment->paymentmethod;
 
-							if ( $transaction_id ) {
+							if ( $transaction_id  && preg_match('/^(?:paypal|paypal[a-zA-Z]+)$/i',$paymentmethod)) {
```
Comment (jlivas) [inline connect.php:1598]: I think we can loosen this regex to just look for anything with 'paypal' in the string

File: connect.php [hunk 4]
```diff
@@ -1652,8 +1653,9 @@ public function sync() {
...
```
Comment (jlivas) [inline connect.php:1658]: I think we can loosen this regex to just look for anything with 'paypal' in the string

**Review by jlivas — CHANGES_REQUESTED** [review]

File: connect.php (hunk 4 above)
Comment (bhushan-mehta) [reply connect.php]: DONE

File: connect.php (hunk 3 above)
Comment (bhushan-mehta) [reply connect.php]: DONE

**Pull request merged**